### PR TITLE
For review

### PR DESCRIPTION
**Pull request by Aleksandr64**

### CODE REVIEW

File: MyBooks/Controllers/AuthorsController.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public IActionResult AddAuthor([FromBody] AuthorVm author)
             return Ok();
         }
 
-        [HttpGet("AuthorAndBooks")]
+        [HttpGet("get-authors-with-books-by-id")]
```
Comment (Lyub4i) [inline MyBooks/Controllers/AuthorsController.cs]: You need to write path to your endpoints without -. It'll be as "GetAuthorBooksById"

File: MyBooks/Controllers/PublishersController.cs [hunk 2]
```diff
@@ -20,5 +20,12 @@ public IActionResult AddNewPublisher([FromBody] PublisherVm publisher)
             _publisherServices.AddPublisher(publisher);
             return Ok();
         }
+
+        [HttpGet("get-publisher-books-with-authore-by-id")]
```
Comment (Lyub4i) [inline MyBooks/Controllers/PublishersController.cs]: Here you need to change too

File: Application/ViewModels/BookAuthorsVM.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.ViewModels
+{
+    public class BookAuthorsVM
+    {
+        public string BookName { get; set; }
+        public List<string> BookAuthors { get; set;}
```
Comment (Lyub4i) [inline Application/ViewModels/BookAuthorsVM.cs]: MSDN recommends to use IEnumurable

File: Application/ViewModels/PublisherWithBooksAndAuthorsVM.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.ViewModels
+{
+    public class PublisherWithBooksAndAuthorsVM
+    {
+        public string Name { get; set; }
+        public List<BookAuthorsVM> BookAuthors { get; set; }
```
Comment (Lyub4i) [inline Application/ViewModels/PublisherWithBooksAndAuthorsVM.cs]: Here too

File: Application/Services/PublishersService.cs [hunk 5]
```diff
@@ -27,6 +27,19 @@ public void AddPublisher(PublisherVm publisher)
             _context.Publishers.Add(_publisher);
             _context.SaveChanges();
         }
-
+        public PublisherWithBooksAndAuthorsVM GetPublisherData(int publisherId)
```
Comment (Lyub4i) [inline Application/Services/PublishersService.cs:42]: Firstly, this class will be a repository. But we'll change it after you'll write all code by your course

**Review by Lyub4i — CHANGES_REQUESTED** [review]
Correct all what I wrote

File: Application/ViewModels/BookAuthorsVM.cs (hunk 3 above)
Comment (Aleksandr64) [reply Application/ViewModels/BookAuthorsVM.cs]: Better replace List on IEnumereble?

File: MyBooks/Controllers/PublishersController.cs (hunk 2 above)
Comment (Aleksandr64) [reply MyBooks/Controllers/PublishersController.cs]: What i need change?

File: Application/ViewModels/BookAuthorsVM.cs (hunk 3 above)
Comment (Lyub4i) [reply Application/ViewModels/BookAuthorsVM.cs]: Yes, It's best practices

File: MyBooks/Controllers/PublishersController.cs (hunk 2 above)
Comment (Lyub4i) [reply MyBooks/Controllers/PublishersController.cs]: path to your endpoint

File: MyBooks/Controllers/PublishersController.cs [hunk 6]
```diff
@@ -22,10 +22,18 @@ public IActionResult AddNewPublisher([FromBody] PublisherVm publisher)
...
         [HttpGet("get-publisher-books-with-authore-by-id")]
-        public IActionResult GetPublisherData(int publisherId) 
+        public IActionResult GetPublisherData(int id) 
         {
-            var _responce = _publisherServices.GetPublisherData(publisherId);
+            var _responce = _publisherServices.GetPublisherData(id);
             return Ok(_responce);
         }
+
+        [HttpDelete("delete-publisher-by-id")]
```
Comment (Lyub4i) [inline MyBooks/Controllers/PublishersController.cs]: I asked change it: [HttpDelete("delete-publisher-by-id")]. It's path to your endpoint. Change it in all places

**Pull request merged**